File: guides/java/kora-java-guide-dependency-injection-introduction-app/src/main/java/ru/tinkoff/kora/guide/dependencyinjection/EmailNotifier.java
package ru.tinkoff.kora.guide.dependencyinjection;

import ru.tinkoff.kora.common.Component;
import ru.tinkoff.kora.common.Tag;

@Tag(EmailTag.class)
@Component
public final class EmailNotifier implements Notifier {

    private final MessageFormatter formatter;

    public EmailNotifier(MessageFormatter formatter) {
        this.formatter = formatter;
    }

    @Override
    public String channel() {
        return "email";
    }

    @Override
    public String notifyUser(String message) {
        return "EMAIL: " + formatter.format(message);
    }
}

File: guides/kotlin/kora-kotlin-guide-dependency-injection-introduction-app/src/main/kotlin/ru/tinkoff/kora/guide/dependencyinjection/EmailNotifier.kt
package ru.tinkoff.kora.guide.dependencyinjection

import ru.tinkoff.kora.common.Component
import ru.tinkoff.kora.common.Tag

@Tag(EmailTag::class)
@Component
class EmailNotifier(
    private val formatter: MessageFormatter
) : Notifier {

    override fun channel(): String = "email"

    override fun notifyUser(message: String): String = "EMAIL: ${formatter.format(message)}"
}
